sd_graymatterseg_new

SaraDupont opened the pull request:
Related to issue #453
ornlm denoising should be included in the dependences

jcohenadad commented on scripts/sct_utils.py:
```diff
@@ -62,9 +64,8 @@ def run_old(cmd, verbose=1):
 
 
-def run(cmd, verbose=1):
-    if verbose==2:
```
why removing this?

jcohenadad commented:
please use sct_apply_transfo instead of function in msct_gm_utils

SaraDupont commented:
I didn't use sct_apply_transfo yet but the bug we spotted yesterday is fixed : the tests are running fine in my computer : we should test on yours.
Travis is still failing because of the "import ornlm" ...

jcohenadad commented:
it's now working on my computer.

jcohenadad commented:
Sara, there is a problem with integrity test on Linux. See here:
https://travis-ci.org/neuropoly/spinalcordtoolbox/jobs/78592313

I suggest you use ferguson (or a VM) to solve the issue. More info for testing stations here:
https://github.com/neuropoly/spinalcordtoolbox/wiki/testing#testing-computers-at-neuropoly

thanks

SaraDupont commented:
OK I'm on it.

Sara

2015-09-03 11:09 GMT-04:00 Julien Cohen-Adad <notifications@github.com>:

> Sara, there is a problem with integrity test on Linux. See here:
> https://travis-ci.org/neuropoly/spinalcordtoolbox/jobs/78592313
>
> I suggest you use ferguson (or a VM) to solve the issue. More info for
> testing stations here:
>
> https://github.com/neuropoly/spinalcordtoolbox/wiki/testing#testing-computers-at-neuropoly
>
> thanks
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/neuropoly/spinalcordtoolbox/pull/482#issuecomment-137479611>
> .
>

SaraDupont commented:
On Ferguson: the result is 'almost' the same as the gold standard, except
for some voxel values.
 For 4 of the 5 slices, the voxel values differences are around 10^-7, I'll
add a threshold to the difference we tolerate for those slices.
For 1 slice the maximum voxel values difference is around 0.1. Is that
considered as a problem of integrity?


Sara

2015-09-03 11:24 GMT-04:00 Sara Dupont <sara.m.dupont@gmail.com>:

> OK I'm on it.
>
> Sara
>
> 2015-09-03 11:09 GMT-04:00 Julien Cohen-Adad <notifications@github.com>:
>
>> Sara, there is a problem with integrity test on Linux. See here:
>> https://travis-ci.org/neuropoly/spinalcordtoolbox/jobs/78592313
>>
>> I suggest you use ferguson (or a VM) to solve the issue. More info for
>> testing stations here:
>>
>> https://github.com/neuropoly/spinalcordtoolbox/wiki/testing#testing-computers-at-neuropoly
>>
>> thanks
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/neuropoly/spinalcordtoolbox/pull/482#issuecomment-137479611>
>> .
>>
>
>

jcohenadad commented:
"0.1 difference" is difficult to judge-- it depends on the mean:  0.1 difference on a mean of 1000 is negligible, 0.1 difference on a mean of 0.5 is not negligible.

can you show a snapshot with OSX and Linux results side by side, same colorscale, adjusted for intensity so that we can visually appreciate the difference?

thanks

jcohenadad commented:
working on OSX :+1: 
https://travis-ci.org/neuropoly/spinalcordtoolbox/jobs/78613245

SaraDupont commented:
The differences of values are on GM segmentation results going from 0 to 1.
Mean of non zero values in the gold standard (osx res) is 0.48.

Here are results (0 to 1 values) with the fsl red-Yellow color map : (left:
osx result = gold standard, right: linux result)[image: Images
intégrées 1][image:
Images intégrées 2]



Since the difference of intensity values is not obvious on those shots,
next are the same results showing values from 0.4 to 0.7 : it's easier to
see the difference of intensity  (first: osx result = gold standard, second
linux result)

[image: Images intégrées 6][image: Images intégrées 7]


Sara

2015-09-03 13:08 GMT-04:00 Julien Cohen-Adad <notifications@github.com>:

> working on OSX [image: :+1:]
> https://travis-ci.org/neuropoly/spinalcordtoolbox/jobs/78613245
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/neuropoly/spinalcordtoolbox/pull/482#issuecomment-137513439>
> .
>

SaraDupont commented:
The only issue related to this difference of intensity that I can imagine
is when computing the CSA: for the 'problematic' slice,  GM CSA is 12.
28mm^2 for gold standard and 12.91 mm^2 for the linux segmentation result.
I don't think the correction of the template registration would be impacted
so extracting metrics after sct_register_graymatter should give the same
results.

Sara

2015-09-03 13:46 GMT-04:00 Sara Dupont <sara.m.dupont@gmail.com>:

> The differences of values are on GM segmentation results going from 0 to
> 1. Mean of non zero values in the gold standard (osx res) is 0.48.
>
> Here are results (0 to 1 values) with the fsl red-Yellow color map :
> (left: osx result = gold standard, right: linux result)[image: Images
> intégrées 1][image: Images intégrées 2]
>
>
>
> Since the difference of intensity values is not obvious on those shots,
> next are the same results showing values from 0.4 to 0.7 : it's easier to
> see the difference of intensity  (first: osx result = gold standard, second
> linux result)
>
> [image: Images intégrées 6][image: Images intégrées 7]
>
>
> Sara
>
> 2015-09-03 13:08 GMT-04:00 Julien Cohen-Adad <notifications@github.com>:
>
>> working on OSX [image: :+1:]
>> https://travis-ci.org/neuropoly/spinalcordtoolbox/jobs/78613245
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/neuropoly/spinalcordtoolbox/pull/482#issuecomment-137513439>
>> .
>>
>
>

jcohenadad commented:
Sara, I cannot see the images-- 

Ideally we would like to have the same results between OSX and Linux. So what i suggest is that for now you put a threshold, but let's open an issue to address this problem in the near future.

thanks

SaraDupont commented:
The difference came from c3d -resample : on OSX, the edges voxel are set to 0, not on Linux. I added a padding of 1 voxel all around the image before the resampling and a crop after. 
I updated the reference result on sct_testing_data gitHub repository : the tests run without error on Evans AND Ferguson.
